src/osx/cocoa/dataview.mm

```diff
@@ -2500,9 +2344,20 @@ void UpdateWithRow(int row)
     return false;
 }
 
-bool wxCocoaDataViewControl::Update(const wxDataViewItem& WXUNUSED(parent), const wxDataViewItem& item)
+bool wxCocoaDataViewControl::Update(const wxDataViewItem& parent, const wxDataViewItem& item)
 {
-    [m_OutlineView reloadItem:[m_DataSource getDataViewItemFromBuffer:item]];
+    if (GetSortingColumn())
+    {
+        if (parent.IsOk())
+            [m_OutlineView reloadItem:[m_DataSource getDataViewItemFromBuffer:parent] reloadChildren:YES];
+        else
+            [m_OutlineView reloadData];
+    }
+    else
```

Comment:
This change seems unrelated, so I'm going to undo it, please let me know if I'm missing something here and it's really needed.